python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -260,9 +260,10 @@ def _partial_fit(self,
             X = cupyx.scipy.sparse.coo_matrix((data, (rows, cols)),
                                               shape=X.shape)
         else:
-            X = input_to_cupy_array(X, order='K').array
+            X = input_to_cupy_array(X, order='K',
+                                    check_dtype=[cp.float32, cp.float64]).array
 
-        y = input_to_cupy_array(y).array
+        y = input_to_cupy_array(y, check_dtype=[cp.int32, cp.int64]).array
```

Comment:
Perhaps we could convert it for the user (optionally?). Recently I found it particularly annoying that `X` and `y` couldn't just be of the same dtype, what do you think?

Reply:
That sounds fair to me. I'll update to do it.